chore: manual QA checklist (#3048)
SwissDataScienceCenter/renku: qa/README.md

    4. Ensure said sesssion can be accessed
    5. Ensure the Renku CLI is there
    6. Modify one of the environment files (e.g. `requirements.txt`) and run `renku save`
    7. Shut down the session, launch a new one from the new commit and make sure the package you added is there
  </details>

- [ ] Delete an existing project (UI)
  <details>
    <summary>Details</summary>

    1. Create a new project in the website
    2. Wait fo the image build and KG integration to complete
    3. Delete the project
    4. Ensure the project is truly gone
  </details>

- [ ] Fork a project (UI)
  <details>
    <summary>Details</summary>

    1. Create a new public project in the website
    2. Wait fo the image build and KG integration to complete
    3. Log in as a different user
    4. Fork the project
    5. Ensure that KG integration succeeds in the fork
    6. Ensure that sessions can be launched in the fork
  </details>

## Datasets

- [ ] Create a dataset from the UI
  <details>
    <summary>Details</summary>

    1. Create a new Renku project in the website
    2. Download a dataset from https://zenodo.org/
    3. Upload the data and create the dataset from the UI

>     4. Ensure that the download completed

Reviewer: ```suggestion
    4. Ensure that the upload completed
```

since above we say "upload the data" ?

Author: I fully reworded this seciont

    5. Ensure that large files are in Git LFS

>     6. After uploading the dataset try to clone the git project locally (does this take a really long time)

Author: It is easier to focus the tests on the web ui. We can add additional tests of the CLI. Mixing the two in the same tests can be confusing.

    7. Launch a session in the project
    8. Is the dataset available?
    9. Run `renku dataset ls` can you see the dataset you uploaded?
  </details>

- [ ] Create a dataset with the CLI from a running session
  <details>
    <summary>Details</summary>

    1. Start a renku session
    2. Navigate to https://zenodo.org/
    3. Pick a random dataset and import it in your session with `renku dataset import <zenodo-url>`
    4. Run `git push`
    5. Ensure the dataset was downloaded successfully
    6. Are large files checked in LFS?
    6. Navigate to the "Datasets" section of the project on the website, can you see the dataset you imported?
  </details>

## Workflows

- [ ] Create a workflow file in a session
  <details>
    <summary>Details</summary>
    
    To be filled in
  </details>
  
## Templates

> - [ ] Create a project and start a session

Reviewer: What's the difference between this and `Create a new project on Renku (UI)` higher up?

Reviewer 2: @Panaetius, I understand the difference with the higher up is that here we aim to test all available default templates work properly (line 100) (which makes sense to me).

Author: The point is to test all templates that have not been touched with the other tests above. I will clarify.

# QA Checklist

Name: <Name>

Date: <yyyy-mm-dd>

Renku helm chart version: <0.25.0>

## Projects

- [ ] Create a new project on Renku (UI)
  <details>
    <summary>Details</summary>

    1. Create a new project in the website
    2. Ensure the KG integration is active
    3. Ensure a session can be successfully started on the project

>     4. Ensure said sesssion can be accessed

Reviewer 3: ```suggestion
    4. Ensure said sesssion can be accessed
    5. Ensure the Renku CLI is there
    6. Modify one of the environment files (e.g. `requirements.txt`) and run `renku save`
    7. Shut down the session, launch a new one from the new commit and make sure the package you added is there
```

  </details>

- [ ] Delete an existing project (UI)
  <details>
    <summary>Details</summary>

    1. Create a new project in the website
    2. Wait fo the image build and KG integration to complete
    3. Delete the project
    4. Ensure the project is truly gone
  </details>

- [ ] Fork a project (UI)
  <details>
    <summary>Details</summary>

    1. Create a new public project in the website
    2. Wait fo the image build and KG integration to complete
    3. Log in as a different user
    4. Fork the project
    5. Ensure that KG integration succeeds in the fork
    6. Ensure that sessions can be launched in the fork
  </details>

## Datasets

- [ ] Create a dataset from the UI
  <details>
    <summary>Details</summary>

    1. Create a new Renku project in the website
    2. Download a dataset from https://zenodo.org/

>     3. Upload the data and create the dataset from the UI

Reviewer 3: If we are downloading data from zenodo, why not test the import functionality? 

If testing dataset creation with uploaded data, just find some cat photos online and upload them

Author: I was thinking that the data found on Zenodo is more representative of a research/data science dataset. A few cat photos are small and easy to handle usually. I think striving to handle any Zenodo file is a good thing. I imagine once in a while someone will pick a file that is too large or that causes problems. With cat photos it is more likely to be trivial. But yeah cat photos is fine too. Dogs too 😸

Reviewer 3: yeah maybe - but then if we're just going to zenodo to get some data (and not using the zenodo import functionality) then we should just put some large-ish files somewhere for people to grab. Finding something usable/useful in zenodo can be hit-or-miss in my experience.

    4. Ensure that the download completed
    5. Ensure that large files are in Git LFS
    6. After uploading the dataset try to clone the git project locally (does this take a really long time)
    7. Launch a session in the project
    8. Is the dataset available?
    9. Run `renku dataset ls` can you see the dataset you uploaded?
  </details>

- [ ] Create a dataset with the CLI from a running session
  <details>
    <summary>Details</summary>

    1. Start a renku session
    2. Navigate to https://zenodo.org/

>     3. Pick a random dataset and import it in your session with `renku dataset import <doi-from-zenodo>`

Reviewer 3: ```suggestion
    3. Pick a random dataset and import it in your session with `renku dataset import <zenodo-url>`
```

    4. Run `git push`
    5. Ensure the dataset was downloaded successfully
    6. Are large files checked in LFS?
    6. Navigate to the "Datasets" section of the project on the website, can you see the dataset you imported?

>     7. Is the 

Reviewer 3: ```suggestion

```

  </details>

## Sessions

> - [ ] Autosaves

Reviewer 3: do we want this here now? Hopefully it will change in the next ~2 sprints?

Author: I will remove it Rok. I was thinking the same thing that maybe we should not add it at all.

    5. Close the session
    6. Ensure the session shut down
    7. When you go to start a new session again the UI should tell you that there is an autosave
    8. Check the autosave branch in Gitlab the large file should be in LFS
    9. Launch a session
    10. When the session starts the unsaved data from steps 2, 3 and 4 should be all there in the same state (i.e. some files should
      be committed and unpushed and some files should be uncommited)
    11. Ensure that the large 1GB file is not in LFS (the autosaves added it to LFS not the user so when we recover we should
      put things exactly as they were before).
    12. If the autosave was successfully recovered ensure that the autosave branch is not there in Gitlab and that the 
      UI does not report an autosave branch when you go to start a new session
  </details>

## Workflows

- [ ] Create a workflow file in a session
  <details>
    <summary>Details</summary>
    
    To be filled in
  </details>
  
## Templates

- [ ] Create a project and start a session
  <details>
    <summary>Details</summary>
    

>     1. Pick a template
>     2. Create a project
>     3. Wait for the image to build
>     4. Ensure the image built
>     5. Start a session
>     6. Ensure the session can be accessed
>     7. Repeat for all templates

Reviewer 3: ```suggestion
    1. Pick a template
    2. Create a project
    3. Wait for the image to build
    4. Ensure the image built
    5. Start a session
    6. Ensure the session can be accessed
    7. Ensure the renku CLI is installed and working
    8. Repeat for all templates
```
